helper/resource/testing_test.go

```diff
@@ -874,6 +874,87 @@ func TestCheckResourceAttr_empty(t *testing.T) {
 	}
 }
 
+func TestTestRangeLengthResourceAttr(t *testing.T) {
+	resName := "test_resource"
+	resKey := "test_key"
+	resVal := "test_value"
+
+	s := terraform.NewState()
+	s.AddModuleState(&terraform.ModuleState{
+		Path: []string{"root"},
+		Resources: map[string]*terraform.ResourceState{
+			resName: {
+				Primary: &terraform.InstanceState{
+					Attributes: map[string]string{
+						resKey: resVal,
+					},
+				},
+			},
+		},
+	})
+
+	t.Run("in_range", func(t *testing.T) {
+		check := TestRangeLengthResourceAttr(resName, resKey, 2, 20)
+		if err := check(s); err != nil {
+			t.Fatal(err)
+		}
+	})
+	t.Run("out_of_range", func(t *testing.T) {
+		check := TestRangeLengthResourceAttr(resName, resKey, 20, 40)
+		if err := check(s); err == nil {
+			t.Fatal(fmt.Errorf("failed to detect attribute '%s.%s' value '%s' length is out of range",
+				resName,
+				resKey,
+				resVal))
+		}
+	})
+}
+
+func TestTestMatchLengthResourceAttr(t *testing.T) {
+	resName := "test_resource"
+	resKey := "test_key"
+	resVal := "test_value"
+
+	s := terraform.NewState()
+	s.AddModuleState(&terraform.ModuleState{
+		Path: []string{"root"},
+		Resources: map[string]*terraform.ResourceState{
+			resName: {
+				Primary: &terraform.InstanceState{
+					Attributes: map[string]string{
+						resKey: resVal,
+					},
+				},
+			},
+		},
+	})
+
+	t.Run("matches_length", func(t *testing.T) {
+		check := TestMatchLengthResourceAttr(resName, resKey, 10)
```

Comment:
Would it make more sense to refer to the [length](https://pkg.go.dev/unicode/utf8#RuneCountInString) of the `resVal` string rather than hardcoding 10?

Reply:
Well, it's on purpose: given that this test is especially focused on "string lengths", keeping it hardcoded makes it break if anything changes, like the string length.

As you can see below, the length becomes part of the errors we expect to see (i.e. `11 == string is too short` and `2 == string is too long`).